language/stdlib/modules/validator_config_2.move

```diff
@@ -0,0 +1,142 @@
+    // if so, the updated key will make it to the ValidatorSet on
+    // a regular reconfiguration when all validators' configs get copied to ValidatorSet.
+    resource struct ValidatorConfigUpdateToken {
+        addr: address,
+        sequence_number: u64,
+    }
+    // Getters for ValidatorConfigUpdateToken
+    public fun get_token_address(update_token: &ValidatorConfigUpdateToken): &address { &update_token.addr }    
+    public fun get_token_sequence_number(update_token: &ValidatorConfigUpdateToken): u64 { *&update_token.sequence_number }
+    // Destroy method for ValidatorConfigUpdateToken
+    public fun destroy_update_token(update_token: ValidatorConfigUpdateToken) {
+        let ValidatorConfigUpdateToken { addr: _, sequence_number: _ } = update_token;
+    }
+    
+    // Returns true if addr has a published ValidatorConfig::T resource
+    public fun has(addr: address): bool { exists<T>(addr) }
+
+    // Getters for Config
+    public fun get_config(addr: address): Config acquires T { *&borrow_global<T>(addr).config }
+    public fun get_consensus_pubkey(config_ref: &Config): vector<u8> { *&config_ref.consensus_pubkey }
+    public fun get_sequence_number(config_ref: &Config): u64 { *&config_ref.sequence_number }
+
+    // Register the transaction sender as a candidate validator by creating a ValidatorConfig
+    // resource under their account. Note that only one such resource can be instantiated under an account.
+    public fun register_candidate_validator(
+        consensus_pubkey: vector<u8>,) {
+
+        move_to_sender<T>(
+            T {
+                config: Config {
+                    consensus_pubkey: consensus_pubkey,
+                    sequence_number: 0,
+                },
+                managing_delegated: false,
+                delegated_address: 0x0
+            }
+        );
+    }
+
+    // If the sender of the transaction has a ValidatorConfig::T resource,
+    // this function will delegate management of the ValidatorConfig::T resource to a delegated_address.
+    public fun set_delegated_address(delegated_address: address) acquires T {
+        Transaction::assert(LibraAccount::exists(delegated_address), 5);
+        let t_ref = borrow_global_mut<T>(Transaction::sender());
+        *(&mut t_ref.managing_delegated) = true;
+        *(&mut t_ref.delegated_address) = delegated_address;
+    }
+
+    // If the sender of the transaction has a ValidatorConfig::T resource,
+    // this function will revoke management capabilities from the delegated_address account.
+    public fun remove_delegated_address() acquires T {
+        let t_ref = borrow_global_mut<T>(Transaction::sender());
+        *(&mut t_ref.managing_delegated) = false;
+        *(&mut t_ref.delegated_address) = 0x0;
+    }
+
+    // Getters for delegated information.
+    // Those getters are only callable by the validator's account owner,
+    // others should be oblivious to how this account is managed.
```

Comment:
I don't think this really matters, since it will be accessible outside of move anyway

Reply:
You are right, I'll delete for now.